src/test/java/core/basesyntax/service/operationwithdata/FruitOperationHandlerTest.java

```diff
@@ -0,0 +1,43 @@
+package core.basesyntax.service.operationwithdata;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.dto.FruitDto;
+import core.basesyntax.fruitoperationstrategy.FruitStrategy;
+import core.basesyntax.fruitoperationstrategy.FruitStrategyImpl;
+import core.basesyntax.operations.Operation;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FruitOperationHandlerTest {
+    private static final String FRUIT_NAME = "banana";
+    private static final Integer QUANTITY_FIRST = 30;
+    private static final Integer QUANTITY_SECOND = 16;
+    private static OperationHandler operationHandler;
+    private static FruitStrategy fruitStrategy;
+
+    @BeforeClass
+    public static void beforeClass() {
+        Map<Operation, FruitOperationService> fruitOperationServiceMap = new HashMap<>();
+        fruitOperationServiceMap.put(Operation.BALANCE, new AddOperation());
+        fruitOperationServiceMap.put(Operation.RETURN, new AddOperation());
+        fruitOperationServiceMap.put(Operation.PURCHASE, new SubtractOperation());
+        fruitOperationServiceMap.put(Operation.SUPPLY, new AddOperation());
+        fruitStrategy = new FruitStrategyImpl(fruitOperationServiceMap);
+        operationHandler = new FruitOperationHandler(fruitStrategy);
+    }
+
+    @Test
+    public void operationProcessing_Ok() {
```

Comment:
what if we purchase more than we have on balance?

Reply:
done